File: sdks/guides/typescript/integrations/llm-anthropic.ts
// Third-party integration - requires: pnpm add @anthropic-ai/sdk
// See: /guides/ai-agents

import Anthropic from '@anthropic-ai/sdk';

const client = new Anthropic();

// > Anthropic usage
export async function complete(messages: Array<{ role: string; content: string }>) {
  const resp = await client.messages.create({
    model: 'claude-3-5-haiku-20241022',
    max_tokens: 1024,
    messages: messages.map((m) => ({ role: m.role as 'user' | 'assistant', content: m.content })),
  });
  const toolUse = resp.content.find((b) => b.type === 'tool_use');
  if (toolUse && toolUse.type === 'tool_use') {
    return {
      content: '',
      toolCalls: [{ name: toolUse.name, args: toolUse.input }],
      done: false,
    };
  }
  const text = resp.content
    .filter((b): b is { type: 'text'; text: string } => b.type === 'text')
    .map((b) => b.text)
    .join('');
  return { content: text, toolCalls: [], done: true };
}
// !!

File: examples/python/guides/integrations/llm_anthropic.py
# Third-party integration example - requires: pip install anthropic
# See: /guides/ai-agents

import json
from anthropic import Anthropic

client = Anthropic()


# > Anthropic usage
def complete(messages: list[dict]) -> dict:
    resp = client.messages.create(
        model="claude-3-5-haiku-20241022",
        max_tokens=1024,
        messages=[{"role": m["role"], "content": m["content"]} for m in messages],
        tools=[{"name": "get_weather", "description": "Get weather", "input_schema": {"type": "object", "properties": {"location": {"type": "string"}}}}],
    )
    for block in resp.content:
        if block.type == "tool_use":
            return {"content": "", "tool_calls": [{"name": block.name, "args": block.input}], "done": False}
    text = "".join(b.text for b in resp.content if hasattr(b, "text"))
    return {"content": text, "tool_calls": [], "done": True}
